_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,116 @@
+
+<br/>
+<center><img src="/img/tech-debt-quadrant.png" alt="The Martin Fowler Technical Debt Quadrant"/></center>
+
+<center><a href="https://martinfowler.com/bliki/TechnicalDebtQuadrant.html">The Tech Debt Quadrant via Martin Fowler</a></center>
+
+<br/>
+It may be impossible to measure the intangible technical debt, but this property does not always extend to its effects. We can find another measurement (or more) that can act as a suitable proxy. This is not dissimilar to maintenance practices in heavy industry. You can never know for certain when a piece of equipment is going to fail, but you can measure how well maintenance crews are adhering to a planned maintenance schedule. In most cases this is a reasonable indication of equipment health (Weber, A. 2005). What measures could we identify that could act as reasonable proxies for the amount of technical debt across one or more software systems?
+
+If we are to find a substitute measure for technical debt we first need to figure out what attributes it would have. For a measure to be beneficial it must be difficult to exploit for individual gain at the expense of the business. Any metrics we choose should have as few side effects as possible. Ideally such such a metric should have a positive correlation with software best practices. Reducing technical debt via some proxy (whatever form that takes) is not practical if it conflicts with good hygiene (e.g. writing tests).
+
+<br/>
+<center><img src="/img/key-metrics.png" alt="The Four Key Metrics by DevOps Research and Assessment"/></center>
+
+<center><a href="https://cloud.google.com/devops">The Four Key Metrics via DORA</a></center>
+
+
+<br/>
+I believe a suitable proxy for technical debt can be found in existing literature on software delivery. The State of DevOps Report (Forsgren, N. et al. 2018) produced by DevOps Research and Assessment (hereby refered to as "the DORA Report") is compiled annually and details the habits of high performing software teams. It is one of the most rigourous studies on software development, produced from surveying and observing hundreds of teams across the world. If we are to come upon a satisfactory measure of technical debt, it would likely be in the DORA report. The report distinguishes two key measures of software delivery performance; throughput and stability. Throughput is measured via deployment frequency and lead time for changes. Stability is measured via time to restore a service and the change failure rate. I believe that change lead time and time to restore service are useful signals of the level of technical debt a project has acrued. Both of these marry well with the effects of technical debt discussed earlier which included decreased velocity and increasingly risky changes.
```

Comment:
are useful signals of the level of technical debt a project has ~~acrued~~ accrued